### PR TITLE
Allow searching issues by release:latest

### PR DESCRIPTION
**Pull request by dcramer**
@getsentry/infrastructure

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4073)

### CODE REVIEW

File: src/sentry/search/utils.py [hunk 1]
```diff
@@ -18,6 +18,15 @@ class InvalidQuery(Exception):
     pass
 
 
+def parse_release(project, value):
```
Comment (dcramer) [inline src/sentry/search/utils.py]: i was gonna add environment support, but its pretty tricky to query that at the moment, and id rather when we do it we just precompute it

File: src/sentry/search/utils.py [hunk 2]
```diff
@@ -18,6 +18,15 @@ class InvalidQuery(Exception):
...
+    # TODO(dcramer): add environment support
+    if value == 'latest':
```
Comment (mattrobenolt) [inline src/sentry/search/utils.py]: I feel like we should make sure there isn't a Release actually named `latest` first before assuming it's the latest by time.

Comment (dcramer) [inline src/sentry/search/utils.py]: no one should be naming a release latest, that would just be silly

Comment (mattrobenolt) [inline src/sentry/search/utils.py]: Sure, but nothing prevents them. It's just a string. People do weird stuff. :)

File: src/sentry/search/utils.py [hunk 3]
```diff
@@ -223,9 +232,9 @@ def parse_query(project, query, user):
                         # an invalid user is entered
                         results['bookmarked_by'] = User(id=0)
             elif key == 'first-release':
-                results['first_release'] = value
+                results['first_release'] = parse_release(project, value)
```
Comment (mattrobenolt) [inline src/sentry/search/utils.py]: Will this be problematic if the value is `None`?

**Comment by mitsuhiko** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/4073)*

File: src/sentry/search/utils.py (hunk 3 above)
Comment (dcramer) [inline src/sentry/search/utils.py]: resolved

**Comment by dcramer** [comment]
@mitsuhiko I agree that its not ideal in all situations, but it's the best we have ATM

---

Review status: 2 of 5 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/4073#-:-KR-SNp7Fto4xyRCy6BL:b-etij74)*

File: src/sentry/search/utils.py (hunk 2 above)
Comment (mattrobenolt) [inline src/sentry/search/utils.py]: Yeah, I agree with this. I'd rather reject at creation time rather than query for it's existence here.

**Comment by mattrobenolt** [comment]
Reviewed 1 of 3 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/4073)*

**Pull request merged**